gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/FakeStorageRpc.java

```diff
@@ -0,0 +1,249 @@
+    }
+    // TODO: crc, etc
+    return object;
+  }
+
+  @Override
+  public Tuple<String, Iterable<Bucket>> list(Map<Option, ?> options) throws StorageException {
+    return null;
+  }
+
+  @Override
+  public Tuple<String, Iterable<StorageObject>> list(String bucket, Map<Option, ?> options)
+      throws StorageException {
+    if (!options.isEmpty()) throw new NotImplementedException();
+    return null;
+  }
+
+  /**
+   * Returns the requested bucket or {@code null} if not found.
+   *
+   * @throws StorageException upon failure
+   */
+  @Override
+  public Bucket get(Bucket bucket, Map<Option, ?> options) throws StorageException {
+    if (!options.isEmpty()) throw new NotImplementedException();
+    return null;
+  }
+
+  /**
+   * Returns the requested storage object or {@code null} if not found.
+   *
+   * @throws StorageException upon failure
+   */
+  @Override
+  public StorageObject get(StorageObject object, Map<Option, ?> options) throws StorageException {
+    if (!options.isEmpty()) throw new NotImplementedException();
+    String key = fullname(object);
+    if (stuff.containsKey(key)) {
+      StorageObject ret = stuff.get(key);
+      if (contents.containsKey(key)) {
+        ret.setSize(BigInteger.valueOf(contents.get(key).length));
+      }
+      return ret;
+    }
+    return null;
+  }
+
+  @Override
+  public Bucket patch(Bucket bucket, Map<Option, ?> options) throws StorageException {
+    if (!options.isEmpty()) throw new NotImplementedException();
+    return null;
+  }
+
+  @Override
+  public StorageObject patch(StorageObject storageObject, Map<Option, ?> options)
+      throws StorageException {
+    if (!options.isEmpty()) throw new NotImplementedException();
+    return null;
+  }
+
+  @Override
+  public boolean delete(Bucket bucket, Map<Option, ?> options) throws StorageException {
+    return false;
+  }
+
+  @Override
+  public boolean delete(StorageObject object, Map<Option, ?> options) throws StorageException {
+    String key = fullname(object);
+    contents.remove(key);
+    return null != stuff.remove(key);
+  }
+
+  @Override
+  public BatchResponse batch(BatchRequest request) throws StorageException {
+    return null;
+  }
+
+  @Override
+  public StorageObject compose(Iterable<StorageObject> sources, StorageObject target,
+                               Map<Option, ?> targetOptions) throws StorageException {
+    return null;
+  }
+
+  @Override
+  public byte[] load(StorageObject storageObject, Map<Option, ?> options) throws StorageException {
+    String key = fullname(storageObject);
+    if (!contents.containsKey(key)) {
+      throw new StorageException(404, "File not found: " + key);
+    }
+    return contents.get(key);
+  }
+
+  @Override
+  public Tuple<String, byte[]> read(
+      StorageObject from, Map<Option, ?> options, long zposition, int bytes)
+      throws StorageException {
+    if (!options.isEmpty()) throw new NotImplementedException();
+    long position = zposition;
```

Comment:
because the linter asked me to, because I modify position later. I can revert if you like.